benchmarks/run.sh

```diff
@@ -1,6 +1,7 @@
-#!/usr/bin/env bash
+#!/usr/bin/sh
 
-cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+benchmark_dir=$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)
+cd "$benchmark_dir"
```

Comment:
Damn, this is uglier than I thought it would be. Just for my own learning, why isn't replacing `${BASHSOURCE[0]}` with `$0` sufficient?

Reply:
Admittedly, I simply used what is suggested in https://stackoverflow.com/a/29835459/704831 (I used the first version, we don't need to resolve the full path). I don't really get why `> /dev/null` is replaced by `CDPATH= …`, but both are not really pretty.

Reply:
Same here, the explanation about `CDPATH=` in the SO answer isn't very clear.

Anyway, I think a clarifying comment about what the line does, along with a link to the SO answer, would be great.